test/apps/iodemo/Makefile.am

```diff
@@ -9,15 +9,28 @@ bin_PROGRAMS = io_demo
 noinst_HEADERS = \
 	ucx_wrapper.h
 
+
+if HAVE_CUDA
+io_demo_CUDA_LDFLAGS = $(CUDA_LDFLAGS)
+io_demo_CUDA_LIBS = $(CUDA_LIBS)
+io_demo_CUDA_CPPFLAGS = $(CUDA_CPPFLAGS) -DHAVE_CUDA
+else
+io_demo_CUDA_LDFLAGS =
+io_demo_CUDA_LIBS =
+io_demo_CUDA_CPPFLAGS =
+endif
+
+
 io_demo_CXXFLAGS = \
 	$(BASE_CXXFLAGS)
 
-io_demo_CPPFLAGS = $(BASE_CPPFLAGS)
-io_demo_LDFLAGS  = -ldl
+io_demo_CPPFLAGS = $(BASE_CPPFLAGS) $(io_demo_CUDA_CPPFLAGS)
+io_demo_LDFLAGS  = -ldl $(io_demo_CUDA_LDFLAGS)
```

Comment:
no need special vars for io_demo_CUDA_LDFLAGS etc
just use `io_demo_LDFLAGS += ...` under `if HAVE_CUDA` / `endif`